### PR TITLE
Move internal features out of app module

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205077183964550/f

### Description
Move the internal features out of `:app` module

Had to create a `:common-ui-internal` module to move the `ThemesPreviewInternalFeature` in there and incidentally re-org'ed the `common` modules

### Steps to test this PR
smoke test internal features.
verify internal features don't appear in play build

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3362** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3362" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3362?utm_source=stack-comment).

File: common/common-ui-internal/build.gradle [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+plugins {
+    id 'com.android.library'
+    id 'kotlin-android'
+    id 'com.squareup.anvil'
+}
+
+apply from: "$rootProject.projectDir/gradle/android-library.gradle"
+
+android {
+    lintOptions {
```
Comment (marcosholgado) [inline common/common-ui-internal/build.gradle:26]: Do you need a lint baseline?

File: common/common-ui-internal/lint-baseline.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<issues format="6" by="lint 7.2.2" type="baseline" client="gradle" dependencies="false" name="AGP (7.2.2)" variant="all" version="7.2.2">
```
Comment (marcosholgado) [inline common/common-ui-internal/lint-baseline.xml:2]: We should be able to delete this as per the above comment

**Review by marcosholgado — APPROVED** [review]
Working as expected. Left a comment re a baseline that can be removed in the common-ui-internal module. Also we could probably migrated `InternalSyncFeature` to sync although we would have to create sync-internal, etc so whatever you want.

**Review by nalcalag — APPROVED** [review]
LGTM! 💯 It would be great if we can move theme-preview classes to this new `common-ui-internal` module at some point.

**Pull request merged**